src/users/users.service.ts

```diff
@@ -70,6 +71,20 @@ export class UsersService {
     return this.usersRepository.findOne({ email });
   }
 
+  async checkAdminPermissions(email: string): Promise<permissionDto> {
+    return {
+      isAdmin: await this.usersRepository
+        .findOne({ email })
+        .then(({ isAdmin }) => {
+          if (isAdmin) {
+            return isAdmin;
+          } else {
+            throw new ForbiddenException('Forbidden');
+          }
+        }),
+    };
```

Comment:
There're several issues:
1. Use async/await instead of `then/catch`

```suggestion
    const user = await this.usersRepository.findOne({ email })
   
    return new PermissionDto({
      isAdmin: user. isAdmin
    });
```
2. Return an exact instance of `PermissionDto` (see the suggestion above)

```ts
class PermissionDto {
  // ... fields

  constructor(params: Partial<PermissionDto>) {
    Object.assign(this, params);
  }
}
```
3. Extract AC logic to middleware or guard

```ts
import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
import { Observable } from 'rxjs';

@Injectable()
export class AdminGuard implements CanActivate { 
  constructor(private readonly users: UsersService) {}

  canActivate(
    context: ExecutionContext,
  ): boolean | Promise<boolean> | Observable<boolean> {
    const permissions = await this.users.checkAdminPermissions();
    
    return permissions.isAdmin;
  }
}
```